providers/base/bin/mouse_keyboard.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+#
+# This file is part of Checkbox.
+#
+# Copyright 2016 Canonical Ltd.
+#
+# Authors:
+#   Gabriel Chen <gabriel.chen@canonical.com>
+#
+# Checkbox is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# Checkbox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox.  If not, see <http://www.gnu.org/licenses/>.
+"""mouse_key_random utility."""
+
+
+from evdev import UInput, ecodes as e
+import time
+import random
+
+# Define keyboard keys and mouse buttons
+KEYBOARD_KEYS = [
+    e.KEY_A,
+    e.KEY_B,
+    e.KEY_C,
+    e.KEY_D,
+    e.KEY_E,
+    e.KEY_F,
+    e.KEY_G,
+    e.KEY_H,
+    e.KEY_I,
+    e.KEY_J,
+    e.KEY_K,
+    e.KEY_L,
+    e.KEY_M,
+    e.KEY_N,
+    e.KEY_O,
+    e.KEY_P,
+    e.KEY_Q,
+    e.KEY_R,
+    e.KEY_S,
+    e.KEY_T,
+    e.KEY_U,
+    e.KEY_V,
+    e.KEY_W,
+    e.KEY_X,
+    e.KEY_Y,
+    e.KEY_Z,
+    e.KEY_1,
+    e.KEY_2,
+    e.KEY_3,
+    e.KEY_4,
+    e.KEY_5,
+    e.KEY_6,
+    e.KEY_7,
+    e.KEY_8,
+    e.KEY_9,
+    e.KEY_0,
+]
+MOUSE_BUTTONS = [e.BTN_LEFT, e.BTN_RIGHT]
+
+
+# Initialize the virtual input device
+def dev_init(name):
+    # Define the capabilities of the device (keyboard and mouse events)
+    capabilities = {
+        e.EV_KEY: KEYBOARD_KEYS + MOUSE_BUTTONS,
+        e.EV_REL: [e.REL_X, e.REL_Y],
+    }
+    # Create the virtual input device
+    device = UInput(
+        capabilities, name=name, vendor=0xBAD, product=0xA55, version=777
+    )
+    time.sleep(1)  # Give userspace time to detect the new device
+    return device
+
+
+# Destroy the virtual input device
+def dev_deinit(device):
+    time.sleep(1)  # Give userspace time to read the remaining events
+    device.close()
+
+
+# Simulate a key press
+def key_press(device, key):
+    device.write(e.EV_KEY, key, 1)  # Press the key
+    device.syn()  # Synchronize the event
+    device.write(e.EV_KEY, key, 0)  # Release the key
+    device.syn()  # Synchronize the event
```

Comment:
```suggestion
    device.write(e.EV_KEY, key, 1)  # Press the key
    device.write(e.EV_KEY, key, 0)  # Release the key
    device.syn()  # Synchronize the event
```

I don't think the first synchronize even is required, [the tutorial](https://python-evdev.readthedocs.io/en/latest/tutorial.html#injecting-input) shows one sync event is enough (and I've checked on my side).